src/core/thread/topology.hpp

```diff
@@ -659,8 +659,23 @@ class Neighbor : public InstanceLocatorInit
     void SetEnhAckProbingMetrics(const otLinkMetrics &aEnhAckProbingMetrics)
     {
         mEnhAckProbingMetrics = aEnhAckProbingMetrics;
-    };
-#endif
+    }
+
+    /**
+     * This method indicates if Enh-ACK Probing is configured and active for this `Neighbor` object.
+     *
+     * @retval TRUE   Enh-ACK Probing is configured and active for this `Neighbor`.
+     * @retval FALSE  Otherwise.
+     *
+     */
+    bool IsEnhAckProbingActive() const
+    {
+        return (mEnhAckProbingMetrics.mPduCount != 0) || (mEnhAckProbingMetrics.mLqi != 0) ||
```

Comment:
I guess `mPduCount` check can be removed since it's not allowed for enh-ack probing.
`mReserved` check could be conditioned for reference device only.

Reply:
Actually `mReserved` is also unnecessary since the Subject only accepts metrics that are supported. This field is used for public API to let Initiator to configure some metrics with reserved field for reference devices.